apps/auth/stats.py

```
@@ -0,0 +1,30 @@
+                "timestamp": log["timestamp"],
+                "status_code": log.get("status_code", None),
+            })
+        return {
+            "total_count": total_count,
+            "logs": log_entries,
+        }
+    except Exception as e:
+        raise HTTPException(
+            status_code=500,
+            detail=f"Error retrieving statistics: {str(e)}",
+        )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)